Header component dropdown

arathjz opened the pull request:
**What:**
- Create a header component dropdown
- Create a header component collapsed for mobile
- Support logo and logoSmall as props for the header to use a custom logo (the reason to add this is due to this stencil issue: https://github.com/ionic-team/stencil/issues/1868, when attempting to load the assets from stencil, it tries to reach the assets from the local path of the project instead of the assets from the `dist` folder, which causes not loading the logo, see next attachment)
![image](https://user-images.githubusercontent.com/20747535/93541828-1468c600-f91d-11ea-88dd-4514da54b7b7.png)


**Why:**
Relates to: https://app.asana.com/0/1159164196409129/1193523850025722/f

**How:**
- Create dropdown to use it on desktop
- Create collapsed to use it on mobile


#### Media:

https://www.loom.com/share/88ed7a9f417e4844a4eb612852700661

duranmla reviewed (approved):
While it looks good I am a bit confused with the decision about having an `<li>` with `onClick` event and the anchor within. Why is this needed?

Also, in the video is not explicit if you find or not a way to make this fully accessible.

And finally, I think the dropdown should have some shadow behind to make it feel as it is on top of things _(like the header is)_.  [Take a look to this screenshot](https://user-images.githubusercontent.com/1425162/93590758-60fce180-f9af-11ea-9aa3-568ca4435b17.png)


Either way, we can move forward if you need to but if we realise there is pending work we don't want to address now please create tickets for it. 

Love it! Thanks for the work and attitude to even add some nice animation. 👍

arathjz commented:
@duranmla Thanks for the feedback, I've addressed your comments. For any missing feature (such as the improvement for a11y) I'll create a ticket for it in Asana
